In [19]:
import math
print(f'The value of pi is approximately {math.pi:.3f}.')

The value of pi is approximately 3.142.


In [20]:
#在 Python 中，使用#标记进行单行注释。
#使用"""或'''来进行多行注释。
#多行注释不能嵌套。
#注释不会被 Python 解释器执行

In [21]:
#代码注释最重要的作用就是让读者可以在不读源码的情况下，快速了解一段代码的主要功能。
#注释是提高代码可读性的重要手段。

In [22]:
#使用\加上符号表示对其后的字符进行转义，如\n表示换行，而不是\和n的简单组合。例如，要在字符串中添加制表符，可使用字符组合\t。


符号	说明
\	如果出现在行尾，则表示续行。即一行未完，转到下一行继续写
\'	单引号
\"	双引号
\n	换行符
\t	横向制表符
\r	回车符
\x	十六进制数代表的字符

要防止字符串中使用转义，需要在字符串前面加上r或者R，表示原始（raw）字符串，例如r"data\600519.csv"

标识符（Identifier）是变量和函数的名称。简单理解，标识符就是用户对变量和函数、类、模块等可以起名字的对象命名时所使用的字符串。

Python 标识符由字母、数字和下划线组成，可以是中文字符。
Python 标识符不能以数字开头。
Python 标识符严格区分大小写。
Python 标识符不能包含空格、@、% 以及 $ 等特殊字符。
不能以系统保留关键字（目前共35个，后续随着版本的更新，肯定会增多）作为标识符

值（Value） 是程序运行过程中的基本元素之一，例如1，3.14，"hello"等等都是值。在编程术语中，它们又被叫作字面量（literals）。字面量拥有不同的类型，如1是整型（int），3.14是浮点型（float），"hello"是字符串（string）。

使用“=”为变量赋值，等于号=也被认为赋值操作符。
在 Python 中，无需声明变量类型，解释器会根据变量的值自动判断变量类型。
Python 变量的数据类型可以动态改变。

Python不同行之间通过冒号和缩进来区分代码块之间的层次。行尾的冒号和下一行的缩进，表示下一个代码块的开始，而缩进的结束则表示此代码块的结束。

if语句后面的条件判断表达式使用 : 结束。
可以使用elif、else进行组合判断。
else后不能加判断条件。
if语句可以嵌套。

In [23]:
##if语句后面的条件判断表达式使用 : 结束。
#可以使用elif、else进行组合判断。
#else后不能加判断条件。
#if语句可以嵌套。

In [24]:
sequence = [1, 2, 3, 4, 5]  
for item in sequence:  
    print(item)
    statements=[6, 7, 8, 9, 10]

1
2
3
4
5


In [39]:
# 循环示例：打印1到10的整数  
for i in range(1, 11):  
    print(i)

1
2
3
4
5
6
7
8
9
10


In [25]:
#循环语法
for iterating_var in sequence:
  statements

In [26]:
#案例1：循环处理序列元素


mylist = [1, 2, 3, 4]
for i in mylist:
    print(i)

1
2
3
4


In [27]:
#案例2： 循环处理字典内容
"""在字典中循环时，用 `items()` 方法可同时取出键和对应的值："""
books = {'标题': 'Python一学就会', '作者': '小明'}
for k, v in books.items():
    print(k, v)

标题 Python一学就会
作者 小明


In [28]:
#"""在序列中循环时，用enumerate()函数可以同时取出位置索引和对应的值："""
for i, v in enumerate(['tic', 'tac', 'toe']):
    print(i, v)

0 tic
1 tac
2 toe


In [29]:
#"""同时循环两个或多个序列时，用zip()函数可以将其内的元素一一匹配"""
questions = ['name', 'quest', 'favorite color']
answers = ['lancelot', 'the holy grail', 'blue']
for q, a in zip(questions, answers):
   print('What is your {0}?  It is {1}.'.format(q, a))

What is your name?  It is lancelot.
What is your quest?  It is the holy grail.
What is your favorite color?  It is blue.


In [30]:
#"""range()函数能够指定循环的起始值和结束值，从而让循环体在指定的范围内循环。例如："""
for i in range(10):
    print(i)                    # 0-9
for i in range(1,10):
    print(i)                    # 1-9
for i in range(1,10,2):
    print(i)                    # 1,3,5,7

0
1
2
3
4
5
6
7
8
9
1
2
3
4
5
6
7
8
9
1
3
5
7
9


In [31]:
#当循环体内部出现 continue 声明时，会结束本次循环，跳转到循环体开始位置，开始下一次循环

In [32]:
#使用 break 语句，配合if语句，可以在满足某个条件的时候中断循环。

In [33]:
#pass 语句不执行任何操作。语法上需要一个语句，但程序不实际执行任何动作时，可以使用该语句。

In [34]:
#什么是CSV
#CSV是Comma-Separated Values的缩写，直译为逗号分隔值，这种格式的文件以纯文本形式存储表格数据（数字和文本）。CSV文件多以逗号为字段之间的分隔符，
# 但也可以使用其他的字符作为字段之间的分隔符。例如：1997，Ford，E350，"ac， abs， moon"，3000.00就是一行CSV格式的数据。CSV因其格式简单，且为纯文本，
# 故被广泛采用在不同数据处理工具中。



In [35]:
#使用Pandas提供的read_excel()方法，可以读取xls及xlsx的数据。不过在使用该方法前还需要安装openpyxl包。

In [36]:
#pandas.read_csv方法的filepath_or_buffer参数可以是字符串形式的本地文件地址，还可以是URL。
# 例如：pd.read_csv('https://adamyang.coding.net/p/data/d/data/git/raw/master/csv/movie.csv')将读取在线的数据。

In [37]:
#read_csv方法的常用参数
#sep或delimiter，用于设定文件分隔符。
#header，用作列名的行号，默认为0，csv文件中没有表头时需要将header设为None。
#index_col，用作行索引的列编号。
#names，自定义的列名列表，结合header=None。
#skiprows，需要跳过读取的行数（从文件开始计算），或需要跳过的行号列表（从0开始）。
#nrows，仅读取开始的若干行。
#encoding，指定unicode的文本编码格式。

In [38]:
#Pandas包中提供了pandas.read_csv方法读取CSV文件，读取成功后，该方法返回DataFrame类型的数据。例如：pd.read_csv('data.csv')。